packages/mwp-app-render/src/components/connectedWithMatchMedia.jsx

```diff
@@ -0,0 +1,35 @@
+//@flow
+import * as React from 'react';
+import { connect } from 'react-redux';
+import type { MapStateToProps, ConnectedComponentClass } from 'react-redux';
+import { createStructuredSelector, createSelector } from 'reselect';
+import withMatchMedia from 'meetup-web-components/lib/utils/components/withMatchMedia';
+
+const mapStateToProps: MapStateToProps<*, *, *> = (state: MWPState) => ({
+	media: state.config.media,
+});
+
+/**
+ * connectedWithMatchMediaComponent
+ * HOC that provides an `initialMedia` prop based on `media` in state.config which is determined by user agent
+ * because withMatchMedia cannot detect the media until after initial render.
+ *
+ * `media` prop is provided from redux state in config.
+ */
+const connectedWithMatchMedia = <Props: {}>(
```

Comment:
I'm not sure what the `<Props: {}>` does on this line - is it a way of declaring a type inline? I've only seen something like `type Props = Object;` before.

I've mainly used this as a reference https://flow.org/en/docs/react/components/#toc-stateless-functional-components

Reply:
yes it's to declare a type inline. I was using this as a reference: https://flow.org/en/docs/react/hoc/

Reply:
Interesting. I'm finding that doc a little hard to follow, but it looks like maybe the typedef for the result would be

```
ConnectedComponentClass<$Diff<Props, { initialMedia: MatchMedia }>, *>
```

and `ConnectedWithMatchMedia` would be implemented with something like `const ConnectedWithMatchMedia = (props: Props) => (` although I'm not 100% sure